### PR TITLE
add OkHttp interceptor to monitor all API calls

### PR DESCRIPTION
**Pull request by dush1729**
Screenshot 2020-01-08 at 4 06 23 PM

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -72,6 +72,9 @@ dependencies {
 
     implementation "org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.2"
     implementation "org.jetbrains.kotlinx:kotlinx-coroutines-android:1.3.2"
+
+    //OkHttp
```
Comment (yevhenii-kanivets) [inline app/build.gradle]: IMHO, this comment is redundant.

File: app/src/main/java/com/bogdan/codeforceswatcher/network/RestClient.kt [hunk 2]
```diff
@@ -5,12 +5,18 @@ import com.google.gson.GsonBuilder
...
 import retrofit2.Retrofit
 import retrofit2.converter.gson.GsonConverterFactory
+import okhttp3.logging.HttpLoggingInterceptor
 
 object RestClient {
 
     private val retrofit by lazy {
         val okHttpClient = OkHttpClient.Builder()
-            .build()
+        val httpLoggingInterceptor = HttpLoggingInterceptor()
```
Comment (yevhenii-kanivets) [inline app/src/main/java/com/bogdan/codeforceswatcher/network/RestClient.kt]: This code can be improved in few ways:
1. Add interceptor only for debug builds (`BuildConfig.DEBUG)` (so we aren't spamming with network logs in production).
2. Use `apply` directly on `HttpLoggingInterceptor`, like this:

```
okHttpClient.addInterceptor(HttpLoggingInterceptor().apply { level = HttpLoggingInterceptor.Level.BODY })
```

File: app/src/main/java/com/bogdan/codeforceswatcher/network/RestClient.kt [hunk 3]
```diff
@@ -21,7 +27,7 @@ object RestClient {
 
         return@lazy Retrofit.Builder()
             .baseUrl("http://www.codeforces.com/api/")
-            .client(okHttpClient)
+            .client(okHttpClient.build())
```
Comment (yevhenii-kanivets) [inline app/src/main/java/com/bogdan/codeforceswatcher/network/RestClient.kt]: If `build()` is called only here, then `okHttpClient` isn't really OkHttpClient anymore, but rather its build. So it's better to call build when initializing `OkHttpClient` in the first place.

**Comment by dush1729** [comment]
Sure. Will raise a issue first. :)

I have resolved the comments. Please check.
Thanks

File: app/src/main/java/com/bogdan/codeforceswatcher/network/RestClient.kt [hunk 4]
```diff
@@ -1,16 +1,22 @@
...
 object RestClient {
 
     private val retrofit by lazy {
         val okHttpClient = OkHttpClient.Builder()
-            .build()
+        if (BuildConfig.DEBUG) {
+            okHttpClient.addInterceptor(HttpLoggingInterceptor()
+                    .apply { level = HttpLoggingInterceptor.Level.BODY })
+        }
+        val okHttpClientBuild = okHttpClient.build()
```
Comment (yevhenii-kanivets) [inline app/src/main/java/com/bogdan/codeforceswatcher/network/RestClient.kt]: Actually this one is `okHttpClient`.

File: app/src/main/java/com/bogdan/codeforceswatcher/network/RestClient.kt [hunk 5]
```diff
@@ -1,16 +1,22 @@
...
 import com.google.gson.GsonBuilder
 import okhttp3.OkHttpClient
 import retrofit2.Retrofit
 import retrofit2.converter.gson.GsonConverterFactory
+import okhttp3.logging.HttpLoggingInterceptor
 
 object RestClient {
 
     private val retrofit by lazy {
         val okHttpClient = OkHttpClient.Builder()
```
Comment (yevhenii-kanivets) [inline app/src/main/java/com/bogdan/codeforceswatcher/network/RestClient.kt]: And this one is `okHttpClientBuilder`.

**Comment by dush1729** [comment]
Done!

**Pull request merged**